pkg/types/types.go

```diff
@@ -84,38 +130,60 @@ type ResourcePool interface {
 	GetMounts(deviceIDs []string) []*pluginapi.Mount
 }
 
-// PciNetDevice provides an interface to get device specific information
-type PciNetDevice interface {
-	GetPFName() string
+// DeviceProvider provides interface for device discovery
+type DeviceProvider interface {
+	AddTargetDevices([]*ghw.PCIDevice, int) error
+	GetDevices() []PciDevice
+}
+
+// PciDevice provides an interface to get generic device specific information
+type PciDevice interface {
 	GetPfPciAddr() string
 	GetVendor() string
 	GetDriver() string
 	GetDeviceCode() string
 	GetPciAddr() string
-	GetNetName() string
 	IsSriovPF() bool
-	GetLinkSpeed() string
-	GetLinkType() string
 	GetSubClass() string
 	GetDeviceSpecs() []*pluginapi.DeviceSpec
```

Comment:
We could tackle this with some additional comments instead of renaming. Renaming this will affect about 55 files.

Reply:
I agree it would touch many files. I think that's also the reason it may cause confusion when reading the code because there are so many files using this function that may come from different interfaces.